lib/rust/parser/src/syntax/token/operator.rs

```diff
@@ -171,9 +167,8 @@ impl HasOperatorProperties for variant::AssignmentOperator {
 impl HasOperatorProperties for variant::TypeAnnotationOperator {
     fn operator_properties(&self) -> OperatorProperties {
         OperatorProperties {
-            binary_infix_precedence: Some(Precedence { value: 2 }),
+            binary_infix_precedence: Some(Precedence::TypeAnnotation),
             lhs_section_termination: Some(SectionTermination::Reify),
-            is_right_associative: true,
```

Comment:
I have it! This is the line that changes the behavior! The rests are just refactorings.

Reply:
This and the precedence change--`:` and `=` both had precedence 2, and now they have different levels in the named level order.